### ISSUE TITLE
Script compile.php is not compatible with PHP 5

### ISSUE THREAD

**Issue by JirkaW** [adminerevo/adminerevo#83]
Script adminer\drivers\mysql.inc.php uses null coalescing operator (??) on lines 1095 and 1101 (add in PHP 7.0).

**Comment by LionelLaffineur** [comment]
Hmm, it is getting more and more difficult to support so old version of PHP.
https://www.php.net/supported-versions.php
PHP 5.x is even not on the supported versions of PHP, even not in end of life.

**Comment by ldav1s** [comment]
I see EOL dates at https://www.php.net/eol.php

The most recent date I see here for PHP 5.x is for [5.6.40](https://www.php.net/releases/#5.6.40) EOL on 31 Dec 2018 (or 10 Jan 2019), not quite 5 years ago.

I don't see any value continuing to support PHP 5.x.  If someone wants that, they can always use adminer.

**Comment by Letme** [comment]
I agree with dropped php5 support.

**Comment by mckaygerhard** [comment]
ok i can confirm the error, it was using php 5.6.40 with alpine linux 3.8 

```
[Thu Jan 09 17:49:44.033062 2025] [proxy_fcgi:error] [pid 4037:tid 3078651220] [client 10.10.34.160:42080] AH01071: Got error 'PHP message: PHP Parse error:  syntax error, unexpected '?' in /usr/share/webapps/adminer/adminer-4.8.4.php on line 65\n'
```

precisely on  `PHP Parse error:  syntax error, unexpected '?' in adminer/include/functions.inc.php on line 502\n'` and also `adminer\drivers\mysql.inc.php`   those uses null coalescing operator (??) on lines 1095 and 1101  a featured included in php 7.0 due the proposal from #153  about https://www.php.net/manual/en/migration70.new-features.php#migration70.new-features.null-coalesce-op for both the offending commits are 385112a and 6283155 that cannot be reverted at all, but with easy patchs can be addressed as suggested in #153

event do `$offendinvar8 = $offendingvalue ?? null` we can easy do `$offendinvar8 = isset($offendingvalue)`


The php 5.6 are still widely used with many servers that's why several hosting services provides .. there are php 5.6 versions for debian extend support in frexian (paid of course)

**Comment by mariancerny** [comment]
AdminerEvo uses null coalescing operator on much more places, just try to find `??` in the source code.

I recommend to switch to this fork: https://github.com/pematon/adminer
It should by compatible with PHP 5.6 up to the latest version 4.12.

**Comment by mckaygerhard** [comment]
yeah @mariancerny  thre are much more places.. i was preparing a pull request but you got me by surprise.. 

interesting.. @mariancerny  what about merge forces and work as team both sides? i mean you know.. here those guys need some help like @LionelLaffineur  ..
